src/Aspire.Dashboard/Extensions/StringExtensions.cs

```diff
@@ -49,4 +31,29 @@ static bool IsValidHtmlIdCharacter(char c)
             return char.IsLetterOrDigit(c) || c == '_' || c == '-';
         }
     }
+
+    /// <summary>
+    /// Returns the two initial letters of the first and last words in the specified <paramref name="name"/>.
+    /// If only one word is present, a single initial is returned. If <paramref name="name"/> is null, empty or
+    /// white space only, <paramref name="defaultValue"/> is returned.
+    /// </summary>
+    public static string? GetInitials(this string name, string? defaultValue = default)
+    {
+        if (string.IsNullOrWhiteSpace(name))
+        {
+            return defaultValue;
+        }
+
+        var initials = name!.Split(" ", StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
```

Comment:
I wouldn't expect the suppression to be needed here:

```suggestion
        var initials = name.Split(" ", StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
```

Reply:
I didn't expect it either, but for some reason it was complaining. It also didn't show the `NotNullWhen` annotation on IsNullOrWhiteSpace when I F12'd to it. So I must have gotten into a bad state somehow. Removed it now and it's not complaining and I see the annotation so I guess something changed in the interim.